Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/DriverOnboarding/VehicleRegistrationCertificate.hs

```diff
@@ -249,30 +255,29 @@ verifyRCFlow person merchantOpCityId imageExtraction rcNumber imageId dateOfRegi
             updatedAt = now
           }
 
-onVerifyRC :: Domain.IdfyVerification -> Idfy.RCVerificationOutput -> Flow AckResponse
-onVerifyRC verificationReq output = do
-  person <- Person.findById verificationReq.driverId >>= fromMaybeM (PersonNotFound verificationReq.driverId.getId)
-  if verificationReq.imageExtractionValidation == Domain.Skipped
-    && compareRegistrationDates output.registration_date verificationReq.issueDateOnDoc
-    then IVQuery.updateExtractValidationStatus verificationReq.requestId Domain.Failed >> return Ack
+onVerifyRC :: Person.Person -> Maybe Domain.IdfyVerification -> VT.RCVerificationResponse -> Flow AckResponse
+onVerifyRC person mbVerificationReq output = do
+  if maybe False (\req -> req.imageExtractionValidation == Domain.Skipped && compareRegistrationDates output.registrationDate req.issueDateOnDoc) mbVerificationReq
+    then IVQuery.updateExtractValidationStatus (maybe "" (. requestId) mbVerificationReq) Domain.Failed >> return Ack
```

Comment:
I think we can't write like this because our usecase is different here,  we need to return ack here.. Also if we case match on `mbVerificationReq` our code will be redundant .

I can write like this:
```
case mbVerificationReq of
  Just req ->
    if req.imageExtractionValidation == Domain.Skipped && compareRegistrationDates output.registrationDate req.issueDateOnDoc
    then IVQuery.updateExtractValidationStatus req.requestId Domain.Failed >> return Ack
    else elsePart
  Nothing -> elsePart

where
  elsePart = do
    -- additional code that runs in the 'else' case
```